macOS/DuckDuckGo/Updates/AppStore/AppStoreOpener.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  AppStoreOpener.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import AppKit
+
+/// Protocol for opening the App Store
+protocol AppStoreOpener {
+    func openAppStore()
+}
+
+/// Default implementation that opens the macOS App Store
+final class DefaultAppStoreOpener: AppStoreOpener {
+    func openAppStore() {
+        NSWorkspace.shared.open(.appStore)
```

Comment:
Just wondering if we need to have any error handling here, as there might be edge cases where App Store is disabled? E.g. https://www.techlockdown.com/guides/disable-app-store-mac

Reply:
I do not think it is necessary. If the user blocks the App Store in the Screen Time, the App Store will still open but it will show the Screen Time screen, also we do not get a callback when that happen.